flask_resources/serializers/base.py

```diff
@@ -46,38 +43,109 @@ def __init__(
+        schema_kwargs = schema_kwargs or {}
+        self.format_serializer = format_serializer_cls(**serializer_options)
+        self.object_schema = object_schema_cls(context=schema_context, **schema_kwargs)
+        if list_schema_cls:
+            self.list_schema = list_schema_cls(
+                context={
+                    "object_schema_cls": object_schema_cls,
+                    **self.schema_context,
+                }
+            )
+        else:
+            self.list_schema = None
 
     def dump_obj(self, obj):
         """Dump the object using object schema class."""
-        return self.object_schema_cls(context=self.schema_context).dump(obj)
+        return self.object_schema.dump(obj)
 
     def dump_list(self, obj_list):
         """Dump the list of objects."""
-        ctx = {
-            "object_schema_cls": self.object_schema_cls,
-        }
-        ctx.update(self.schema_context)
-
-        if self.list_schema_cls is None:
-            return self.object_schema_cls(context=self.schema_context).dump(
-                obj_list, many=True
-            )
+        if not self.list_schema:
+            return self.object_schema.dump(obj_list, many=True)
 
-        return self.list_schema_cls(context=ctx).dump(obj_list)
+        return self.list_schema.dump(obj_list)
 
     def serialize_object(self, obj):
         """Dump the object using the serializer."""
-        return self.format_serializer_cls.serialize_object(self.dump_obj(obj))
+        return self.format_serializer.serialize_object(self.dump_obj(obj))
 
     def serialize_object_list(self, obj_list):
         """Dump the object list using the serializer."""
-        return self.format_serializer_cls.serialize_object_list(
-            self.dump_list(obj_list)
-        )
+        return self.format_serializer.serialize_object_list(self.dump_list(obj_list))
+
+
+class DumperMixin:
+    """Abstract class that defines an interface for pre_dump and post_dump methods.
+
+    It allows to extend records serialization.
+    """
+
+    def post_dump(self, data, original=None, **kwargs):
+        """Hook called after the marshmallow serialization of the record.
+
+        :param data: The dumped record data.
+        :param original: The original record data.
+        :param kwargs: Additional keyword arguments.
+        :returns: The serialized record data.
+        """
+        return data
+
+    def pre_dump(self, data, original=None, **kwargs):
+        """
+        Hook called before the marshmallow serialization of the record.
+
+        :param data: The record data to dump.
+        :param original: The original record data.
+        :param kwargs: Additional keyword arguments.
+        :returns: The data to dump.
+        """
+        return data
+
+
+# FIXME: This should be moved to the base transformer class on
+# https://github.com/inveniosoftware/flask-resources/issues/117
+
+
+class BaseSerializerSchema(Schema):
```

Comment:
This is 100% not ideal, but is the only way to move to this module the processing pipelines. When https://github.com/inveniosoftware/flask-resources/issues/117 is addressed this will be moved to the transformer class.